analyses/tmb-compare/compare-tmb-calculations.Rmd

```diff
@@ -0,0 +1,242 @@
+
+```{r}
+all_tmb <- tmb_ccdl_maftools %>% 
+  dplyr::inner_join(tmb_ccdl_focr, by = "Tumor_Sample_Barcode", 
+                    suffix = c("_maftools", "_focr")) %>% 
+  dplyr::inner_join(tmb_d3b, by = "Tumor_Sample_Barcode") %>% 
+  dplyr::rename(tmb_d3b = TMB, 
+                mutation_count_d3b = count)
+```
+
+## Fraction differences in TMB
+
+Here we use the CCDL's maftools-defined filtered TMB for the denominator since its the current TMB. 
+
+Calculate fraction differences in relation to maftools-defined filtered TMB. 
+
+```{r}
+all_tmb <- all_tmb %>% 
+  dplyr::mutate(frac_diff_focr = (tmb_maftools - tmb_focr) / tmb_maftools, 
+                frac_diff_d3b = (tmb_maftools - tmb_d3b) / tmb_maftools)
+```
+
+Let's plot the distribution of fraction differences from both d3b and focr-defined TMBs. 
+
+```{r}
+all_tmb %>% 
+  dplyr::select(dplyr::starts_with("frac_diff")) %>% 
+  tidyr::gather("tmb_calculation_set", "frac_diff") %>%
+  dplyr::mutate(tmb_calculation_set = gsub("frac_diff_", "", tmb_calculation_set)) %>%
+  ggplot2::ggplot(ggplot2::aes(x = frac_diff, color = tmb_calculation_set)) + 
+  ggplot2::geom_density() + 
+  ggplot2::theme_classic() + 
+  ggplot2::ggtitle("Fraction difference in TMB in comparison to CCDL maftools-defined TMB")
+```
+
+Most of the D3b TMBs are very different. 
+Most of the FoCR-defined TMB are not too different but there's some with big differences.
+
+We'll investigate these a bit. 
+Let's identify which samples have the biggest differences for FoCR-defined TMB.
+
+```{r}
+all_tmb %>% 
+  dplyr::filter(abs(frac_diff_focr) > .05) %>% 
+  dplyr::arrange(desc(frac_diff_focr))
+```
+
+TMB is very different between CCDL and D3b calculations fraction wise. 
+The ranges for the CCDL sets and the D3b sets are very different.  
+
+```{r}
+cat(
+  " Range for CCDL maftools-defined TMB:", range(all_tmb$tmb_maftools), "\n",
+  "Range for CCDL focr-defined TMB:", range(all_tmb$tmb_focr), "\n",
+  "Range for D3b TMB:", range(all_tmb$tmb_d3b)
+  )
+```
+
+Let's split up the TMB fraction to see where this vast difference in ranges might be. 
+
+## How different are mutation counts?
+
+Let's take a look at the numerator of the TMB fraction by selecting the counts columns only. 
+
+```{r}
+counts_only <- all_tmb %>% 
+  dplyr::select(mutation_count_focr, mutation_count_maftools, mutation_count_d3b)
+
+# Print this out
+counts_only
+```
+
+Let's plot these counts correlations, but turn them into log1p values for easier visualization.
+
+```{r}
+counts_only %>%
+  dplyr::transmute_all(log1p) %>%
+  GGally::ggpairs(mapping = ggplot2::aes(alpha = 0.05)) +
+  ggplot2::theme_classic()
+```
+
+The samples with high TMB, still usually have a higher TMB in the other datasets. 
+
+Let's take a closer look at one of the comparisons.
+
+```{r}
+counts_only %>%
+  dplyr::transmute_all(log1p) %>%
+  ggplot2::ggplot(ggplot2::aes(x = mutation_count_maftools, y = mutation_count_d3b)) + 
```

Comment:
```suggestion
  ggplot2::ggplot(ggplot2::aes(x = mutation_count_focr, y = mutation_count_d3b)) + 
```

Reply:
What's your rationale for this? I like comparing to the current TMB which is maftools-defined filter.

Reply:
Aka why do you find it more informative?

Reply:
It is the closest thing we have to an apples-to-apples comparison. These two should be doing the same thing, so differences are due to data or code and can be investigated. Any maftools/focr difference can be explained by the choice of mutations, so that is not very interesting. Maybe include both?

Reply:
I just put both. Why not?